deepchem/rl/a3c.py

```diff
@@ -263,7 +263,7 @@ def create_rollout(self):
     for i in range(self.a3c.max_rollout_length):
       if self.env.terminated:
         break
-      state = self.env.state
+      state = copy.deepcopy(self.env.state)
```

Comment:
This shouldn't be needed.  States should be immutable.  The problem is that you've implemented your environment so that it modifies states after outside code has already retrieved them.  Doing that opens the door to all sorts of bugs in lots of places.  Instead it should assign a new value to `_state` which is then treated as immutable.